pallets/msa/src/types.rs

```diff
@@ -182,6 +184,47 @@ impl AddProvider {
 	}
 }
 
+/// A type definition for the Recovery Commitment data for the following operation:
+/// -  Adding a Recovery Commitment - `pallet_msa::add_recovery_commitment
+pub type RecoveryCommitment = [u8; 32]; // 32 bytes for the recovery commitment
+
+/// A type definition for the payload for the Recovery Commitment operation:
+/// -  Adding a Recovery Commitment - `pallet_msa::add_recovery_commitment`
+#[derive(
+	TypeInfo, RuntimeDebugNoBound, Clone, Decode, DecodeWithMemTracking, Encode, PartialEq, Eq,
+)]
+#[scale_info(skip_type_params(T))]
+pub struct RecoveryCommitmentPayload<T: Config> {
+	/// type discriminator
+	pub discriminant: PayloadTypeDiscriminator,
+	/// The Recovery Commitment (32 bytes)
+	pub recovery_commitment: RecoveryCommitment,
+	/// The block number at which a signed proof of this payload expires.
+	pub expiration: BlockNumberFor<T>,
+}
+
+impl<T: Config> EIP712Encode for RecoveryCommitmentPayload<T> {
+	fn encode_eip_712(&self, chain_id: u32) -> Box<[u8]> {
+		lazy_static! {
+
+			// signed payload
+			static ref MAIN_TYPE_HASH: [u8; 32] = sp_io::hashing::keccak_256(
+				b"RecoveryCommitmentPayload(uint64 msaId,uint32 expiration,bytes32 recoveryCommitment)",
+			);
+		}
+		// get prefix and domain separator
+		let prefix_domain_separator: Box<[u8]> =
+			get_eip712_encoding_prefix("0xcccccccccccccccccccccccccccccccccccccccc", chain_id);
```

Comment:
nit: the `"0xccc...ccc"` value is reused a lot in this file. Should we make a constant?